discovery-infra/test_infra/controllers/node_controllers/libvirt_controller.py

```diff
@@ -50,6 +52,78 @@ def list_nodes_with_name_filter(self, name_filter) -> List[libvirt.virDomain]:
     def list_networks(self):
         return self.libvirt_connection.listAllNetworks()
 
+    @staticmethod
+    def _list_disks(node):
+        all_disks = minidom.parseString(node.XMLDesc()).getElementsByTagName('disk')
+
+        results = []
+
+        for disk_xml in all_disks:
+            results.append(LibvirtController.to_disk_obj(disk_xml))
+
+        return results
+
+    @staticmethod
+    def to_disk_obj(disk_xml):
+        # Indicates how the disk is to be exposed to the guest OS.
+        # Possible values for this attribute are "floppy", "disk", "cdrom", and "lun", defaulting to "disk".
+        device_type = disk_xml.getAttribute('device')
+        alias = LibvirtController._get_disk_alias(disk_xml)
+        wwn = LibvirtController._get_disk_wwn(disk_xml)
+        bus, target = LibvirtController.get_target_data(disk_xml)
+        source_xml = disk_xml.getElementsByTagName('source')
+        source_type = DiskSourceType.OTHER
```

Comment:
Maybe split this function a little
```python
    @classmethod
    def to_disk_obj(cls, disk_xml):
        return Disk(
            **dict(
                # device_type indicates how the disk is to be exposed to the guest OS.
                # Possible values for this attribute are "floppy", "disk", "cdrom", and "lun", defaulting to "disk".
                device_type=disk_xml.getAttribute('device'),
                alias=cls._get_disk_alias(disk_xml),
                wwn=cls._get_disk_wwn(disk_xml),
            ),
            **cls._get_disk_source_attributes(disk_xml),
            **cls._get_disk_target_data(disk_xml)
        )

    @staticmethod
    def _get_disk_source_attributes(disk_xml):
        source_xml = disk_xml.getElementsByTagName('source')

        source_type = DiskSourceType.OTHER
        source_path = None
        source_pool = None
        source_volume = None

        if source_xml:
            disk_type = disk_xml.getAttribute('type')
            source_element = source_xml[0]

            if disk_type == 'file':
                source_type = DiskSourceType.FILE
                source_path = source_element.getAttribute('file')
            elif disk_type == 'block':
                source_type = DiskSourceType.BLOCK
                source_path = source_element.getAttribute('dev')
            elif disk_type == 'dir':
                source_type = DiskSourceType.DIR
                source_path = source_element.getAttribute('dir')
            elif disk_type == 'network':
                source_type = DiskSourceType.NETWORK
            elif disk_type == 'volume':
                source_type = DiskSourceType.VOLUME
                source_pool = source_element.getAttribute('pool')
                source_volume = source_element.getAttribute('volume')
            elif disk_type == 'nvme':
                source_type = DiskSourceType.NVME

        return dict(
            source_type=source_type,
            source_path=source_path,
            source_pool=source_pool,
            source_volume=source_volume,
        )

    @staticmethod
    def _get_disk_target_data(disk_xml):
        target_xml = disk_xml.getElementsByTagName('target')
        return dict(
            bus=target_xml[0].getAttribute('bus') if target_xml else None,
            target=target_xml[0].getAttribute('dev') if target_xml else None
        )
```

Reply:
thanks,  but why the to_disk_obj is a class_method?

Reply:
Because it needs to call class methods